wandb/sdk/launch/loader.py

```diff
@@ -0,0 +1,225 @@
+    # fmt: on
+    raise LaunchError(
+        f"Could not create registry from config. Invalid registry type: {registry_type}"
+    )
+
+
+def builder_from_config(
+    config: Optional[Dict[str, Any]],
+    environment: AbstractEnvironment,
+    registry: AbstractRegistry,
+) -> AbstractBuilder:
+    """Create a builder from a config.
+
+    This helper function is used to create a builder from a config. The
+    config should have a "type" key that specifies the type of builder to import
+    and create. The remaining keys are passed to the builder's from_config
+    method. If the config is None or empty, a DockerBuilder is returned.
+
+    Args:
+        config (Dict[str, Any]): The builder config.
+        registry (Registry): The registry of the builder.
+
+    Returns:
+        The builder.
+
+    Raises:
+        LaunchError: If the builder is not configured correctly.
+    """
+    if not config:
+        # fmt: off
+        from .builder.docker_builder import DockerBuilder
+        return DockerBuilder.from_config({}, environment, registry)  # This is the default builder.
+
+    builder_type = config.get("type")
+    if builder_type is None:
+        raise LaunchError(
+            "Could not create builder from config. Builder type not specified"
+        )
+    # fmt: off
+    if builder_type == "docker":
+        from .builder.docker_builder import DockerBuilder
+        return DockerBuilder.from_config(config, environment, registry)
+    if builder_type == "kaniko":
+        if isinstance(registry, LocalRegistry):
+            raise LaunchError(
+                "Could not create Kaniko builder. "
+                "Registry must be a remote registry."
+            )
+        from .builder.kaniko_builder import KanikoBuilder
+        return KanikoBuilder.from_config(config, environment, registry)
+    if builder_type == "noop":
+        from .builder.noop import NoOpBuilder
+        return NoOpBuilder.from_config(config, environment, registry)
+    # fmt: on
+    raise LaunchError(
+        f"Could not create builder from config. Invalid builder type: {builder_type}"
+    )
+
+
+def runner_from_config(
+    runner_name: str,
+    api: Api,
+    runner_config: Dict[str, Any],
+    environment: AbstractEnvironment,
+) -> AbstractRunner:
+    """Create a runner from a config.
+
+    This helper function is used to create a runner from a config. The
+    config should have a "type" key that specifies the type of runner to import
+    and create. The remaining keys are passed to the runner's from_config
+    method. If the config is None or empty, a LocalContainerRunner is returned.
+
+    Args:
+        runner_name (str): The name of the backend.
+        api (Api): The API.
+        runner_config (Dict[str, Any]): The backend config.
+
+    Returns:
+        The runner.
+
+    Raises:
+        LaunchError: If the runner is not configured correctly.
+    """
+    # fmt: off
```

Comment:
Whats this for?

Reply:
this is to prevent black from putting a ton of empty lines around all the imports in this function. We can totally remove this I just find the blank lines make it the code a bit harder to look at.

Reply:
Let's remove it.